src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -60,104 +62,93 @@ private MsQuicApi(QUIC_API_TABLE* apiTable)
 
     static MsQuicApi()
     {
-        if (!TryLoadMsQuic(out IntPtr msQuicHandle))
+        if (MsQuicHandle == IntPtr.Zero)
+        {
+            // MsQuic library not loaded
+            return;
+        }
```

Comment:
```suggestion
        if (!TryLoadMsQuic(out MsQuicHandle))
        {
            // MsQuic library not loaded
            return;
        }
```
and delete inline initialization of MsQuicHandle above

Reply:
Instead of storing MsQuicHandle, you can consider storing the pointers to the two exports in statics instead. It would save a bit of redundant work on the re-initialization.